### ISSUE TITLE
NDS fails to start on Raspberry Pi 4B

### ISSUE THREAD

**Issue by martignoni** [nodogsplash/nodogsplash#419]
Nodogsplash fails to start on Raspberry Pi 4B (Raspbian Buster, updated to latest package versions).

### Preparation
1. Boot the RPi with a fresh Raspbian lite image (2019-07-10)
1. Launch `sudo apt-get update && sudo apt-get dist-upgrade -y`
1. Install and configure a working wireless AP with `hostapd` and `dnsmasq`
1. Download Nodogsplash source via git and compile it
1. Install the Nodogsplash deb package with `sudo dpkg -i nodogsplash_4.0.1-1_armhf`

### Observed behavior
Following shell message is displayed and Nodogsplash fails to start
```
Job for nodogsplash.service failed because the control process exited with error code.
See "systemctl status nodogsplash.service" and "journalctl -xe" for details.
```

### Expected behavior
Nodogsplash starts up and works correctly.

## Comments
 - Without doing first `sudo apt-get update && sudo apt-get dist-upgrade -y`, the package works flawlessly.
- Tested with version 4.0.1 and 4.0.2, same results for the two versions.
- Impossible to raise the debug level, since NDS fails very early.

**Comment by bluewavenet** [comment]
What did you see in:
`systemctl status nodogsplash.service" and "journalctl -xe" for details.`

**Comment by bluewavenet** [comment]
What settings do you have in the config file?

**Comment by martignoni** [comment]
> What did you see in:
> `systemctl status nodogsplash.service" and "journalctl -xe"`

```
Aug 22 20:13:57 xxx systemd[1]: Starting NoDogSplash Captive Portal...
Aug 22 20:13:57 xxx nodogsplash[1970]: Could not set debuglevel to 5
Aug 22 20:13:57 xxx systemd[1]: nodogsplash.service: Control process exited, code=exited, status=1/FAILURE
Aug 22 20:13:57 xxx systemd[1]: nodogsplash.service: Failed with result 'exit-code'.
Aug 22 20:13:57 xxx systemd[1]: Failed to start NoDogSplash Captive Portal.
Aug 22 20:13:57 xxx systemd[1]: nodogsplash.service: Service RestartSec=100ms expired, scheduling restart.
Aug 22 20:13:57 xxx systemd[1]: nodogsplash.service: Scheduled restart job, restart counter is at 1.
Aug 22 20:13:57 xxx systemd[1]: Stopped NoDogSplash Captive Portal.
```

**Comment by martignoni** [comment]
> What settings do you have in the config file?

```
# Nodogsplash Configuration File
GatewayInterface wlan0
FirewallRuleSet authenticated-users {
    FirewallRule allow all
}
FirewallRuleSet preauthenticated-users {
}
FirewallRuleSet users-to-router {
    FirewallRule allow udp port 53
    FirewallRule allow tcp port 53
    FirewallRule allow udp port 67
    FirewallRule allow tcp port 22
    FirewallRule allow tcp port 80
    FirewallRule allow tcp port 443
}
GatewayName MyHost
GatewayAddress 10.0.0.1
RedirectURL http://myhost.home/
GatewayPort 2050
MaxClients 50
SessionTimeout 360
```

But the same issue occurs with a default configuration!

**Comment by bluewavenet** [comment]
@martignoni

> Aug 22 20:13:57 xxx nodogsplash[1970]: Could not set debuglevel to 5

**From v4.0.0 onwards, valid debug levels are 0, 1, 2 and 3.** This is reflected in the documentation.
https://nodogsplashdocs.readthedocs.io/en/stable/debug.html

Your very short config file does not show where you set to level 5... I assume you have just posted a snippet.

Also you should not be setting GatewayAddress, this should only be necessary in very rare occasions on obscure operating systems or if the NDS router is multi homed, although this should not be causing this issue.

I am not certain, but I think RedirectURL was deprecated somewhere in v 3.x.x as its functionality was included in FAS.

Please, after it has failed, check in /tmp. Does the file ndsctl.lock exist?

**Comment by martignoni** [comment]
@bluewavenet 

> Your very short config file does not show where you set to level 5... I assume you have just posted a snippet.

No this is my complete config file. Moreover, as noted above, with the default config, I get exactly the same symptoms.

However, I'll try your suggestions tonight when I can access to the server (I'm off today).

**Comment by bluewavenet** [comment]
@martignoni
I found the problem. The nodogsplash.service file tries to set debuglevel 5, which is invalid.
For now you will have to edit the file.
Change the line:
`ExecStart=/usr/bin/nodogsplash -d 5 $OPTIONS`
to:
`ExecStart=/usr/bin/nodogsplash -d 1 $OPTIONS`

**Comment by bluewavenet** [comment]
See PR #420

**Comment by bluewavenet** [comment]
Also see pr #421
RedirectURL has not been available in the OpenWrt version since ~v3.1.x
It will still work for you at least up to v4.0.3 although:

> redirectURL is now redundant as most CPD immediately close the "splash" page
> as soon as NDS authenticates, thus redirectURL will not be shown.
> 
> This functionality, ie displaying a particular web page as a final "Landing Page"
> can be achieved reliably using FAS, with NDS calling the previous "redirectURL" as the FAS page.

In the case of MoodleBox you would configure nodogsplash.conf as follows (noting that if the page you are redirecting to is on the same router as NDS, port 80 is reserved by NDS and cannot be used):
```
  fasremotefqdn http://myhost.home/
  fasremoteip [ip_address_of_myhost.home]
  fasport [ip_port_of_myhost.home_server]
```
and if required:
```
fas_secure_enable [0, 1 or 2 ]
faskey [your_secret_aes_key]
```
Full details for setting up FAS can be found in the documentation at:
https://nodogsplashdocs.readthedocs.io/en/stable/fas.html

**Comment by martignoni** [comment]
> ExecStart=/usr/bin/nodogsplash $OPTIONS

I confirm that this fixes the issue. A big **thank you**.
